libs/gl-client/src/scheduler.rs

```diff
@@ -433,10 +434,10 @@ where
     where
         T: GrpcClient,
     {
-        let cert_node_id = self.get_node_id_from_tls_config()?;
+        let node_id = get_node_id_from_tls_config(&self.creds.tls_config())?;
```

Comment:
See the other comment, we may consider to just add the node_id to the credentials metadata as this can only be called by authenticated schedulers.